scripts/prodsec-manifest-generator.sh

```diff
@@ -1,82 +1,131 @@
+        PRODUCT_VERSION_ID=$PRODUCT_NAME:$PRODUCT_VERSION
+        PRODUCT_URL=$(yq r ${FILEPATH} "products[$i].url")
+        PRODUCT_INSTALL_TYPE=$(yq r ${FILEPATH} "products[$i].installType")
+
+        if [[ "$PRODUCT_INSTALL_TYPE" == *"$OLM_TYPE"* ]]; then
+          TMP_DIR=$(mktemp -d)
+          echo "Generating manifest for ${PRODUCT_NAME} - version ${PRODUCT_VERSION}"
+          git clone -c advice.detachedHead=false --quiet --depth 1 -b ${PRODUCT_VERSION} ${PRODUCT_URL} ${TMP_DIR}
+          cd ${TMP_DIR}
+          go mod graph | cut -d " " -f 2 | tr @ - | while read x; do echo "${SERVICE_NAME}:${VERSION}/$PRODUCT_VERSION_ID/$x" >> "$GENERATION_FILE"; done
+          cd ${CURRENT_DIR}
+          rm -rf $TMP_DIR
+        fi
+    done
 
     # Dependencies used
-    go mod graph | cut -d " " -f 2 | tr @ - | while read x; do echo "${SERVICE_NAME}:${VERSION}/$x" >> "$PRE_SORTED_FILE"; done
+    go mod graph | cut -d " " -f 2 | tr @ - | while read x; do echo "${SERVICE_NAME}:${VERSION}/$x" >> "$GENERATION_FILE"; done
 
-    # Remove repeating dependencies
-    sort -u "$PRE_SORTED_FILE" > "prodsec-manifests/$FILE_NAME"
+    cat "$GENERATION_FILE" > "$FILE_NAME"
 
-    # Delete pre-sorted file
-    rm -f "$PRE_SORTED_FILE"
+    echo "Manifest generated successfully, deleting temporary files"
+    rm -f "$GENERATION_FILE"
 }
 
-
+CURRENT_DIR=$(pwd)
 
 case $TYPE_OF_MANIFEST in
 "master")
-    SERVICE_NAME="services-rhoam"
-    VERSION="master"
-    FILE_NAME="rhoam-master-manifest.txt"
-
+    case $OLM_TYPE in
+      "integreatly-operator")
+        OLM_TYPE="rhmi"
+        VERSION="master"
+        FILE_NAME="${CURRENT_DIR}/prodsec-manifests/rhmi-master-manifest.txt"
+        SERVICE_NAME="services-rhmi"
+        ;;
+      "managed-api-service")
+        OLM_TYPE="rhoam"
+        VERSION="master"
+        FILE_NAME="${CURRENT_DIR}/prodsec-manifests/rhoam-master-manifest.txt"
+        SERVICE_NAME="services-rhoam"
+        ;;
+      *)
+        echo "Invalid OLM_TYPE set"
+        echo "Use \"integreatly-operator\" or \"managed-api-service\""
+        exit 1
+        ;;
+    esac
     manifest_generate
-    exit 0
     ;;
 "compare")
-    SERVICE_NAME="services-rhoam"
-    VERSION="master"
-    FILE_NAME="master-from-branch-manifest.txt"
-
-    RHOAM_MASTER_FROM_BRANCH="prodsec-manifests/master-from-branch-manifest.txt"
-    RHOAM_CURRENT_MASTER="prodsec-manifests/rhoam-master-manifest.txt"
-    RHOAM_SORTED_MASTER="prodsec-manifests/rhoam-master-sorted-manifest.txt"
-
+        case $OLM_TYPE in
+      "integreatly-operator")
+        OLM_TYPE="rhmi"
+        VERSION="master"
+        FILE_NAME="${CURRENT_DIR}/prodsec-manifests/master-from-branch-manifest.txt"
+        SERVICE_NAME="services-rhmi"
+        MASTER_FROM_BRANCH="${CURRENT_DIR}/prodsec-manifests/master-from-branch-manifest.txt"
+        SORTED_FROM_BRANCH="${CURRENT_DIR}/prodsec-manifests/sorted-master-from-branch-manifest.txt"
```

Comment:
does this need to be duplicated in the switch case? it seems to have the same values

Reply:
Good point, thanks!